#!/bin/bash
#SBATCH --job-name=MUMer # Job name
#SBATCH --mail-type=END,FAIL
#SBATCH --mail-user=sisi.chen@ufl.edu    # Where to send mail
#SBATCH --cpus-per-task=10
#SBATCH --mem-per-cpu=5g
#SBATCH --time=96:00:00              # Time limit hrs:min:sec
#SBATCH --output=mumer_%j.log # Standard output and error log
#SBATCH --qos=xingbo.wu-b
date;hostname;pwd

module load mummer
#module load syri
#module load python3
#module minimap2

for i in Alanpur Caraboa ordorata Safeda Haden
do
   mkdir $i
    cd $i
    
    ####MUMmer    
    
###Whole genome alignment. Any other alignment can also be used.
nucmer --maxmatch -c 100 -b 500 -l 50 /blue/xingbo.wu/sisi.chen/03.mango/00.data/Chrom_genome/alphonso_Chr.fa /blue/xingbo.wu/sisi.chen/03.mango/00.data/Chrom_genome/"$i"_Chr.fa
####Remove small and lower quality alignments
delta-filter -m -i 90 -l 100 out.delta > "$i"_i90_m_delta.filter
####Convert alignment information to a .TSV format as required by SyRI
show-coords -THrd "$i"_i90_m_delta.filter > "$i"_i90_m_delta.filter.coords

    ####syri

module load minimap2
####Using minimap2 for generating alignment. Any other whole genome alignment tool can also be used.
minimap2 -ax asm5 --eqx /blue/xingbo.wu/sisi.chen/03.mango/00.data/Chrom_genome/alphonso_Chr.fa /blue/xingbo.wu/sisi.chen/03.mango/00.data/Chrom_genome/"$i"_Chr.fa > alphonso_"$i".sam
####syri will output the genome structure difference files syri.out and syri.vcf
module load conda
conda activate syri_env

python3 /apps/syri/1.5.4/bin/syri -c alphonso_"$i".sam -r /blue/xingbo.wu/sisi.chen/03.mango/00.data/Chrom_genome/alphonso_Chr.fa -q /blue/xingbo.wu/sisi.chen/03.mango/00.data/Chrom_genome/"$i"_Chr.fa -k -F S --prefix "$i" 

####Identification of genome rearrangements from whole-genome comparisons generated by MUMmer using SyRI
python3 syri -c "$i"_i90_m_delta.filter.coords -d "$i"_i90_m_delta.filter -r /blue/xingbo.wu/sisi.chen/03.mango/00.data/Chrom_genome/alphonso_Chr.fa -q /blue/xingbo.wu/sisi.chen/03.mango/00.data/Chrom_genome/"$i"_Chr.fa 
####plot the results
python3 plotsr "$i"syri.out /blue/xingbo.wu/sisi.chen/03.mango/00.data/Chrom_genome/alphonso_Chr.fa /blue/xingbo.wu/sisi.    chen/03.mango/00.data/Chrom_genome/"$i"_Chr.fa -H 8 -W 5


cd ..
done
date
